### PR TITLE
Convert issue form to yaml

### PR DESCRIPTION
**Pull request by michelleangela**
The following changes are proposed:
- Converts the issue form to yaml.
- Improve issue reporting tool.

Please provide feedback on any changes to this form, such as wording of descriptions/titles, or any changes to any of the fields.

### CODE REVIEW

**Comment by bobbrow** [comment]
Can you also delete `docs/issue_template.md`?

File: .github/ISSUE_TEMPLATE/bugreport.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    description: |
+      Put a short summary here.
+      If this is a bug, please provide clear instructions as to how we can reproduce the issue on our end.
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: CMake Tools Diagnostics
+    description: |
+      For bugs, please run the `CMake: Log Diagnostics` command and paste the output here.
```
Comment (bobbrow) [inline .github/ISSUE_TEMPLATE/bugreport.yml:21]: Is it possible to tell the form to format this like code \`\`\`?

Comment (bobbrow) [reply .github/ISSUE_TEMPLATE/bugreport.yml:21]: `render: shell` maybe?

Comment (michelleangela) [reply .github/ISSUE_TEMPLATE/bugreport.yml:21]: Made the input for logs to render as shell.

**Comment by michelleangela** [comment]
Would you like the pull request template converted to yaml too?

**Review by bobbrow — APPROVED** [review]

**Comment by bobbrow** [comment]
I'm not sure how much we need the pull request template.  The thing I want most from it is the linked bug number.  I think everyone just deletes the rest.

**Pull request merged**